test/components/views/rooms/wysiwyg_composer/SendWysiwygComposer-test.tsx

```diff
@@ -68,83 +70,119 @@ describe('SendWysiwygComposer', () => {
-        defaultDispatcher.dispatch({
-            action: Action.FocusSendMessageComposer,
-            context: null,
-        });
-
-        // Then the component gets the focus
-        await waitFor(() => expect(screen.getByRole('textbox')).toHaveFocus());
-    });
-
-    it('Should focus and clear when receiving an Action.ClearAndFocusSendMessageComposer', async () => {
-        // Given we don't have focus
-        customRender(jest.fn(), jest.fn());
-        expect(screen.getByRole('textbox')).not.toHaveFocus();
-
-        // When we send the right action
-        defaultDispatcher.dispatch({
-            action: Action.ClearAndFocusSendMessageComposer,
-            context: null,
-        });
-
-        // Then the component gets the focus
-        await waitFor(() => expect(screen.getByRole('textbox')).toHaveFocus());
-        expect(mockClear).toBeCalledTimes(1);
+        // Then
+        expect(screen.getByTestId('WysiwygComposer')).toBeTruthy();
     });
 
-    it('Should focus when receiving a reply_to_event action', async () => {
-        // Given we don't have focus
-        customRender(jest.fn(), jest.fn());
-        expect(screen.getByRole('textbox')).not.toHaveFocus();
+    it('Should render PlainTextComposer when isRichTextEnabled is at false', () => {
+        // When
+        customRender(jest.fn(), jest.fn(), false, false);
 
-        // When we send the right action
-        defaultDispatcher.dispatch({
-            action: "reply_to_event",
-            context: null,
-        });
-
-        // Then the component gets the focus
-        await waitFor(() => expect(screen.getByRole('textbox')).toHaveFocus());
+        // Then
+        expect(screen.getByTestId('PlainTextComposer')).toBeTruthy();
     });
 
-    it('Should not focus when disabled', async () => {
-        // Given we don't have focus and we are disabled
-        customRender(jest.fn(), jest.fn(), true);
-        expect(screen.getByRole('textbox')).not.toHaveFocus();
-
-        // When we send an action that would cause us to get focus
-        defaultDispatcher.dispatch({
-            action: Action.FocusSendMessageComposer,
-            context: null,
-        });
-        // (Send a second event to exercise the clearTimeout logic)
-        defaultDispatcher.dispatch({
-            action: Action.FocusSendMessageComposer,
-            context: null,
+    describe.each([{ isRichTextEnabled: true }, { isRichTextEnabled: false }])(
+        'Should focus when receiving an Action.FocusSendMessageComposer action',
+        ({ isRichTextEnabled }) => {
+            afterEach(() => {
+                jest.resetAllMocks();
+            });
+
+            it('Should focus when receiving an Action.FocusSendMessageComposer action', async () => {
+            // Given we don't have focus
+                customRender(jest.fn(), jest.fn(), false, isRichTextEnabled);
+                document.head.focus();
+                //    screen.getByLabelText('Bold').focus();
```

Comment:
Can you clean up the left overs? :broom:

Reply:
done